src/index.ts

```diff
@@ -53,9 +53,11 @@ function Poke<Body, Result>(host:string, options?:PokeOption<Body>, callback?:(p
                         // emit respnse
                         // NOTE: ts seems not able to infer the result type but it doesn't matter in pratice
                         // NOTE: probably due to the mixing use of genric and typeguard and tsc is not powerful enough for that?
-                        eventManager.response(result as PokeSuccess<Result>)
+                        if (eventManager.response)
```

Comment:
newly added type for Event stuff catched a few bugs in this file, callback can be undefined when they are not yet registered. Need some runtime check for them